src/Scripting/Core/ObjectFormatter.cs

```diff
@@ -27,53 +29,64 @@ public string FormatObject(object obj, ObjectFormattingOptions options = null)
 
         #region Reflection Helpers
 
-        private static bool HasOverriddenToString(Type type)
+        private static bool HasOverriddenToString(TypeInfo type)
         {
-            Ref.MethodInfo method = type.GetMethod(
-                "ToString",
-                Ref.BindingFlags.Public | Ref.BindingFlags.Instance,
-                null,
-                Type.EmptyTypes,
-                null
-            );
-            return method.DeclaringType != typeof(object);
+            if (type.IsInterface)
+            {
+                return false;
+            }
+
+            while (type.AsType() != typeof(object))
+            {
+                if (type.GetDeclaredMethod("ToString", Type.EmptyTypes) != null)
+                {
+                    return true;
+                }
+
+                type = type.BaseType.GetTypeInfo();
+            }
+
+            return false;
         }
 
-        private static DebuggerDisplayAttribute GetApplicableDebuggerDisplayAttribute(Ref.MemberInfo member)
+        private static DebuggerDisplayAttribute GetApplicableDebuggerDisplayAttribute(MemberInfo member)
         {
-            var result = (DebuggerDisplayAttribute)member.GetCustomAttributes(typeof(DebuggerDisplayAttribute), inherit: true).FirstOrDefault();
+            // includes inherited attributes:
+            var result = member.GetCustomAttributes<DebuggerDisplayAttribute>().FirstOrDefault();
             if (result != null)
             {
                 return result;
             }
 
             // TODO (tomat): which assembly should we look at for dd attributes?
-            Type type = member as Type;
+            var type = member as TypeInfo;
             if (type != null)
             {
-                foreach (DebuggerDisplayAttribute attr in type.Assembly.GetCustomAttributes(typeof(DebuggerDisplayAttribute), inherit: true))
+                foreach (DebuggerDisplayAttribute attr in type.Assembly.GetCustomAttributes<DebuggerDisplayAttribute>())
                 {
-                    if (IsApplicableAttribute(type, attr.Target, attr.TargetTypeName))
+                    if (IsApplicableAttribute(type, attr.Target.GetTypeInfo(), attr.TargetTypeName))
                     {
                         return attr;
                     }
                 }
             }
+
             return null;
         }
 
-        private static DebuggerTypeProxyAttribute GetApplicableDebuggerTypeProxyAttribute(Type type)
+        private static DebuggerTypeProxyAttribute GetApplicableDebuggerTypeProxyAttribute(TypeInfo type)
         {
-            var result = (DebuggerTypeProxyAttribute)type.GetCustomAttributes(typeof(DebuggerTypeProxyAttribute), inherit: true).FirstOrDefault();
+            // TODO: inherited?
+            var result = type.GetCustomAttribute<DebuggerTypeProxyAttribute>();
```

Comment:
Yes, DebuggerTypeProxy attribute can be inherited. And DebuggerTypeProxyAttribute.AllowMultiple = true.